hal/mbed_pinmap_common.c

```diff
@@ -37,6 +37,14 @@ void pinmap_pinout(PinName pin, const PinMap *map)
 
 uint32_t pinmap_merge(uint32_t a, uint32_t b)
 {
+    if (
+        (a != (uint32_t)NC)
+        && (b != (uint32_t)NC)
+    ) {
+        // mis-match error case
+        MBED_ERROR1(MBED_MAKE_ERROR(MBED_MODULE_PLATFORM, MBED_ERROR_CODE_PINMAP_INVALID), "pinmap mis-match", a);
```

Comment:
The original error was raised if a != NC && b != NC && a != b
What was the warning raised?

Reply:
Fixed with this [force-push](https://github.com/ARMmbed/mbed-os/compare/104118121a39f76760bd40a63b2c2ba53196e460..f02a4f1c70327e77d522ab53d40114cda9a03c65)
To answer your question, the warning was raised because the last return statement was unreachable.

Reply:
This code would have been better fixed by just removing the return statement. The compiler was merely pointing out that the `MBED_ERROR1` call does not return, so the final return `(uint32_t)NC` was redundant.

Reply:
Done. https://github.com/ARMmbed/mbed-os/compare/f02a4f1c70327e77d522ab53d40114cda9a03c65..682e2ef26319d308a6899dfe4e474034ad35cff2